docker/linux-runner

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+set -e
+
+arch=$1
+prog=$2
+
+case "$CROSS_RUNNER" in
+  native)
+    exec $prog
+    ;;
+  qemu-user)
+    exec qemu-$arch $prog
+    ;;
+  qemu-system)
+    true
+    ;;
+  *)
+    echo "Invalid CROSS_RUNNER: $CROSS_RUNNER";
+    exit 1
+    ;;
+esac
+
+if [ "$arch" = "arm" ]; then
+  opt="-machine virt"
+fi
+
+if [ "$arch" = "aarch64" ]; then
+  opt="-machine virt -cpu cortex-a57"
+fi
+
+(
+  flock -n 200 || exit 0
+
+  qemu-system-$arch \
+    -m 1024 \
+    -smp 4 \
+    -nographic \
+    $opt \
+    -kernel /qemu/vmlinuz \
+    -initrd /qemu/initrd.gz \
+    -serial tcp:127.0.0.1:4444,server,nowait \
+    -virtfs local,path=/target,security_model=mapped,mount_tag=target \
```

Comment:
We use virtfs to share /target with the guest running in qemu. The files do not need to be explicit copied!

Reply:
In `stdsimd` we use `objdump` from a procedural macro to check the assembly generated by SIMD intrinsics. `objdump` isn't typically installed by default in the distros that are usually loaded into `qemu-sytem`.

I think we could hack around this by copying whichever `objdump` we want to use into `target/` and passing its path in an environment variable to qemu system, but maybe there would be a better way of sharing resources and binaries with the code running in `qemu-system` ?

Reply:
qemu-system is used to run the target binaries, the build happens in the docker container, which has `objdump`. Does this makes sense?

Reply:
I think so. So what happens if my target binary tries to start a new process that runs `objdump` ? Does the one from the docker container get picked up?

Reply:
In this case there would be no `objdump`... The qemu-system image is really small, it is basically a busybox rootfs... If this is necessary (sorry if I misunderstood  you), I think you could put extra binaries in /target and use [passthrough](https://github.com/japaric/cross#passing-environment-variables-into-the-build-environemnt) to pass env variable to inform where the binaries are.

Reply:
I see, that's exactly what I meant in my first comment.